Main.hs

```diff
@@ -51,11 +51,10 @@ stringP :: String -> Parser String
 stringP = sequenceA . map charP
 
 jsonBool :: Parser JsonValue
-jsonBool = f <$> (stringP "true" <|> stringP "false")
-   where f "true"  = JsonBool True
-         f "false" = JsonBool False
-         -- This should never happen
-         f _       = undefined
+jsonBool = jsonTrue <|> jsonFalse
+   where mkJsonBool str b = const (JsonBool b) <$> stringP str
```

Comment:
`const` + `<$>` is such a common pattern that there is a shorthand for it `<$`. Try `JsonBool b <$ stringP str`

Reply:
That's pretty cool